core/commands/config.go

```diff
@@ -219,22 +250,20 @@ func getConfig(r repo.Repo, key string) (*ConfigField, error) {
 }
 
 func setConfig(r repo.Repo, key string, value interface{}) (*ConfigField, error) {
-	err := r.SetConfigKey(key, value)
+	keyF, err := getConfig(r, "Identity.PrivKey")
 	if err != nil {
-		return nil, fmt.Errorf("Failed to set config value: %s (maybe use --json?)", err)
+		return nil, errors.New("failed to get PrivKey")
 	}
-	return getConfig(r, key)
-}
-
-func showConfig(filename string) (io.Reader, error) {
-	// TODO maybe we should omit privkey so we don't accidentally leak it?
-
-	data, err := ioutil.ReadFile(filename)
+	privkey := keyF.Value
+	err = r.SetConfigKey(key, value)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to set config value: %s (maybe use --json?)", err)
 	}
-
-	return bytes.NewReader(data), nil
+	err = r.SetConfigKey("Identity.PrivKey", privkey)
```

Comment:
- this code is trying to "set the privatekey again" with the original value, to be sure the user did not change it.
- this is an incorrect way to do this
- and is subject to data races.
- and is subject to "triggered crash" problem (if i can cause a crash _after_ the first write and before the second, i can set it to disk successfully. given i can trigger a crash, i can achieve this by doing it over and over again until i get it right.)
- the right way to do this is to sanitize the input and not let it set `Identity.PrivKey` at all.
- I know it is tricky because the value may be `Identity.PrivKey`, or the whole `Identity` block, or the whole config.
- The "setting it again" thing should not be relied upon for security.
- I would probably "set the private key" into the incoming value.
- Another option is removing `Identity` entirely (not just the private key), after all, the `Identity` is tied to the private key and shouldn't be mutable either...